tools/common/FileCopier.cs

```diff
@@ -123,6 +123,13 @@ static CopyFileResult CopyFileCallback (CopyFileWhat what, CopyFileStep stage, I
 					else if (stage == CopyFileStep.Err) {
 						Log (1, "Could not copy the file '{0}' to '{1}'", source, target);
 						return CopyFileResult.Quit;
+					} else if (stage == CopyFileStep.Start) {
+						if (File.Exists (target) || Directory.Exists (target)) {
+							Log (1, "Deleted target {0}, it's not up-to-date", target);
+							// This callback won't be called for directories, but we can get here for symlinks to directories.
+							// This means that File.Delete should always work (no need to check for a directory to call Directory.Delete)
+							File.Delete (target);
+						}
```

Comment:
```suggestion
						Log (1, "Deleted target {0}, it's not up-to-date", target);
						// This callback won't be called for directories, but we can get here for symlinks to directories.
						// This means that File.Delete should always work (no need to check for a directory to call Directory.Delete)
						File.Delete (target);
```

Reply:
The logging statement can end up confusing now, because it will say it deleted something that may or may not have existed.

Reply:
Is not a big issue, We can leave it as it is, is just that File.Exist + File.Delete might give threading issues, but those File.* methods always do :)